libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/indexes.rs

```diff
@@ -220,4 +220,37 @@ pub(crate) fn fulltext_columns_should_not_use_arguments(
     }
 }
 
+/// The ordering is only possible with `BTree` access method.
+pub(crate) fn hash_index_must_not_use_sort_param(
+    db: &ParserDatabase<'_>,
+    index: IndexWalker<'_, '_>,
+    diagnostics: &mut Diagnostics,
+) {
+    if !db.preview_features.contains(PreviewFeature::ExtendedIndexes) {
+        return;
+    }
+
+    if !db
+        .active_connector()
+        .has_capability(ConnectorCapability::UsingHashIndex)
+    {
+        return;
+    }
```

Comment:
Are these two conditions needed? I assume that without the preview feature, we'll never encounter hash indexes in the schema.

Reply:
I try to avoid error spam, if you don't have them in the connector and/or the flag set, I don't want proper validation errors to appear to things like "can't use this in a hash index".

Reply:
but there won't be hash indexes, so these errors will never appear, right?

Reply:
you can add `type: Hash` and it would have two validation errors: one for using a hash index and another using sort in a hash index